jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -5,16 +6,94 @@ XXX jcssackett 2015-09-18: Licensing for juju.js? It's different then the
 licensing for the GUI.
 */
 
-;
+'use strict';
+
+var module = module;
+
+/**
+ * jujulib provides api access for microservices used by juju.
+ *
+ * jujulib provies access to the APIs for the Juju Environment
+ * Manager (JEM), the juju charmstore, and the juju identity
+ * manager (IdM).
+ */
+(function (exports) {
+
+    /**
+     * Environment object for jujulib.
+     *
+     * Provides access to the JEM API.
+     */
 
-var jujulib = (function (exports) {
+    /**
+     * Initializer
+     *
+     * @function environment
+     * @param url {String} The url, including scheme and port, of the JEM instance.
+     * @param bakery {Object} A bakery object for communicating with the JEM instance.
+     * @returns {Object}
+     */
+    var environment = function(url, bakery) {
+        var jemUrl = url + '/v1';
+
+        /**
+         * Wrapper for making requests via the bakery.
+         *
+         * @private _makeRequest
+         * @param path {String} The JEM endpoint to make the request from,
+         *     e.g. '/env'
+         * @param success {function} A callback to be called on success.
+         * @param failure {function} A callback to be called on failure.
+         */
+        var _makeRequest = function(path, success, failure) {
+            bakery.sendGetRequest(path, null, success, failure);
+        };
+
+        /**
+         * Lists the available environments on the JEM.
+         *
+         * @public listEnvironments
+         * @param success {function} A callback to be called on success. Should
+         *     take an array of environments as its one parameter.
+         * @param failure {function} A callback to be called on failure. Should
+         *     take an error message as its one parameter.
+         */
+        this.listEnvironments = function(success, failure) {
+            _makeRequest(jemUrl + '/env', function(xhr) {
+                var data = JSON.parse(xhr.target.responseText);
```

Comment:
I wonder if we should really create an intermediate object before passing the returned data to callbacks.
I mean, that would decouple the server responses from the internal code execution in this library, and would also result in the callback arguments to be more explicitly defined in the code. That way, we could also easily react to slight changes on the returned server data (yeah, I know that it should never happen...). Same for the API call below.
What do you think?